homeassistant/components/mpd/media_player.py

```diff
@@ -217,8 +217,15 @@ def media_content_type(self):
     @property
     def media_duration(self):
         """Return the duration of current playing media in seconds."""
-        # Time does not exist for streams
-        return self._currentsong.get("time")
+        time_from_status = self._status.get("time")
+        time = None
+        if (
+            time_from_status
+            and isinstance(time_from_status, str)
+            and ":" in time_from_status
+        ):
+            time = time_from_status.split(":")[1]
+        return self._currentsong.get("time", time)
```

Comment:
```suggestion
        time_from_status = self._status.get("time")
        if (
            isinstance(time_from_status, str)
            and ":" in time_from_status
        ):
            return time_from_status.split(":")[1]
        return self._currentsong.get("time")
```
`isinstance(None, str)` is False so we don't have to check for `time_from_status` implicitly.
Maybe using `:=` operator in the `if` would be nice here too (can `time_from_status` be something else than str?)

Reply:
I want the `self._currentsong.get("time")` to be the default so returning inside the if not makes sense. anyway i changed the code to simplify the logic

Reply:
Oh right. So why don't return immediately and do the alternative handling only if this _currentsong is None?
```py
if currentsong_time := self._currentsong.get("time"):
    return currentsong_time
# self._status.get("time") stuff here
```